package/gluon-core/luasrc/lib/gluon/upgrade/820-dns-config

```diff
@@ -9,6 +9,14 @@ uci:set('dhcp', dnsmasq, 'localservice', false)
 
 if site.dns and site.dns.servers then
 	uci:set('dhcp', dnsmasq, 'server', site.dns.servers)
+	-- allow inbound traffic for dns from client zone
+	uci:section('firewall', 'rule', 'clientdns',{
+		name = 'clientdns',
+		src = 'client',
+		dest_port = '53',
+		proto = 'tcpudp',
+		target = 'ACCEPT',
+	})
```

Comment:
Hmm, I'm unsure if we want to start adding firewall rules conditionally. If we do, we should also remove the section in the *else* branch.

Reply:
There is an old rule *client_dns* in `320-gluon-mesh-batman-adv-client-bridge` that should be removed. We could just call this rule *client_dns*, unconditionally remove it above the `if dns.servers then`, and then add it again in the `if dns.servers then`.